src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -17,15 +17,18 @@ public class DefaultViewComponentHelper : IViewComponentHelper, ICanHasViewConte
         private readonly IViewComponentInvokerFactory _invokerFactory;
         private readonly IViewComponentSelector _selector;
         private ViewContext _viewContext;
+        private readonly ILogger _logger;
 
         public DefaultViewComponentHelper(
             [NotNull] IViewComponentDescriptorCollectionProvider descriptorProvider,
             [NotNull] IViewComponentSelector selector,
-            [NotNull] IViewComponentInvokerFactory invokerFactory)
+            [NotNull] IViewComponentInvokerFactory invokerFactory,
+            [NotNull] ILoggerFactory loggerFactory)
         {
             _descriptorProvider = descriptorProvider;
             _selector = selector;
             _invokerFactory = invokerFactory;
+            _logger = loggerFactory.CreateLogger<DefaultViewComponentHelper>();
```

Comment:
You can also just add `ILogger<DefaultViewComponentHelper>` as a constructor parameter instead of taking the factory